packages/template-block/src/TemplateBlock.tsx

```diff
@@ -0,0 +1,218 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import {
+    OpenNewPublicationPayload,
+    Template,
+    openNewPublication,
+    useBlockSettings,
+    useBlockTemplates,
+    useEditorState,
+} from '@frontify/app-bridge';
+import { ReactElement, useCallback, useEffect, useState } from 'react';
+import { BlockProps } from '@frontify/guideline-blocks-settings';
+import { PreviewType, Settings, cardPaddingValues, cornerRadiusValues, textPositioningToFlexDirection } from './types';
+import { GAP, SETTING_ID } from './constants';
+import { Button, ButtonEmphasis, Color, Heading, Text, TextInput, Textarea, merge } from '@frontify/fondue';
+import { getRgbaString } from './utils';
+import { TemplatePreview } from './components/TemplatePreview';
+import { AlertError } from './components/AlertError';
+
+export const TemplateBlock = ({ appBridge }: BlockProps): ReactElement => {
+    const [blockSettings, updateBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const [selectedTemplate, setSelectedTemplate] = useState<Template | null>(null);
+    const [lastErrorMessage, setLastErrorMessage] = useState('');
+    const isEditing = useEditorState(appBridge);
+    const blockId = appBridge.getBlockId();
+    const { blockTemplates, updateTemplateIdsFromKey, error } = useBlockTemplates(appBridge);
+
+    const {
+        title,
+        description,
+        preview,
+        isCardPaddingCustom,
+        cardPaddingSimple,
+        cardPaddingCustomTop,
+        cardPaddingCustomLeft,
+        cardPaddingCustomRight,
+        cardPaddingCustomBottom,
+        hasCardBackgroundColor,
+        cardBackgroundColor,
+        hasCardBorder,
+        cardBorderColor,
+        cardBorderStyle,
+        cardBorderWidth,
+        isCardCornerRadiusCustom,
+        cardCornerRadiusSimple,
+        cardCornerRadiusCustom,
+        textPositioning,
+        textRatio,
+        textAnchoringHorizontal,
+        textAnchoringVertical,
+    } = blockSettings;
+
+    const [templateTitle, setTemplateTitle] = useState(title ?? '');
+    const [templateDescription, setTemplateDescription] = useState(description ?? '');
+
+    const hasPreview = preview !== PreviewType.None;
+    const flexDirection = hasPreview ? textPositioningToFlexDirection[textPositioning] : 'row';
+    const isRows = hasPreview && (flexDirection === 'row' || flexDirection === 'row-reverse');
+
+    const updateTemplateTitle = (value: string) => {
+        setTemplateTitle(value);
+        onChangeSetting('title', value);
+    };
+
+    const updateTemplateDescription = (value: string) => {
+        setTemplateDescription(value);
+        onChangeSetting('description', value);
+    };
+
+    const updateSelectedTemplate = useCallback(
```

Comment:
Instead of this function and the useEffect at line 89, can you just access the template directly from `blockTemplates[SETTINGS_ID]` ?

Reply:
Yep, this code can be simplified. I've cleaned up the `updateSelectedTemplate` function but still I need that `useEffect` hook to process data updates from `useBlockTemplates` custom hook which is basically provides a template data.